src/sentry/static/sentry/app/components/checkboxFancy/checkboxFancy.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import {css} from '@emotion/core';
+
+import {Theme} from 'app/utils/theme';
+
+import CheckboxFancyContent from './checkboxFancyContent';
+
+type Props = {
+  isDisabled?: boolean;
+  size?: string;
+  className?: string;
+} & React.ComponentProps<typeof CheckboxFancyContent>;
+
+const disabledStyles = (p: Props & {theme: Theme}) =>
+  p.isDisabled &&
+  css`
+    background: ${p.isChecked || p.isIndeterminate ? p.theme.gray1 : p.theme.offWhite};
+    border-color: ${p.theme.gray1};
+  `;
+
+const hoverStyles = (p: Props & {theme: Theme}) =>
+  !p.isDisabled &&
+  css`
+    border: 2px solid ${p.isChecked || p.isIndeterminate ? p.theme.purple : p.theme.gray4};
+  `;
+
+const CheckboxFancy = styled(
+  ({isChecked, className, isDisabled, isIndeterminate}: Props) => (
+    <div
+      data-test-id="checkbox-fancy"
+      role="checkbox"
+      aria-disabled={isDisabled}
+      aria-checked={isChecked}
+      className={className}
+    >
+      <CheckboxFancyContent isIndeterminate={isIndeterminate} isChecked={isChecked} />
+    </div>
+  )
+)`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  box-shadow: 1px 1px 5px 0px rgba(0, 0, 0, 0.05) inset;
```

Comment:
Could we put this inset border in theme?

Reply:
hmm, let me think of how to name it and add this change in a follow-up PR 😉